### ISSUE TITLE
VisibilityModifier.allowPublicImmutableFields should allow public immutable fields in enums

### ISSUE THREAD

**Issue by scompt** [checkstyle/checkstyle#2887]
## Code
```java
public enum Blah {
    A, B;
    
    public final int blah = 5;
}
```

## Config
```xml
<?xml version="1.0" encoding="UTF-8"?><!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name="Checker">

    <property name="severity" value="warning"/>

    <module name="TreeWalker">
        <module name="VisibilityModifier">
            <property name="allowPublicImmutableFields" value="true"/>
        </module>
    </module>

</module>
```

## Observed Output
```bash
java -jar checkstyle-6.15-all.jar -c config.xml -f plain Blah.java
Starting audit...
[WARN] Blah.java:4:22: Variable 'blah' must be private and have accessor methods. [VisibilityModifier]
Audit done.
```

## Expected Output
```bash
java -jar checkstyle-6.15-all.jar -c config.xml -f plain Blah.java
Starting audit...
Audit done.
```

**Comment by rnveach** [comment]
@scompt 
> public final int blah = 5;

1) 
https://github.com/checkstyle/checkstyle/blob/ba6781e1098fbbd76482c67a2e21f2796db166a2/src/main/java/com/puppycrawl/tools/checkstyle/checks/design/VisibilityModifierCheck.java#L537
It looks like the check wants the field to be `static final`, not just `final`.
Is there a reason your field isn't static? It is declared `final`, so I don't see it mattering it being static or non-static except for looks as you declare the value on declaration.
2) 
https://github.com/checkstyle/checkstyle/blob/ba6781e1098fbbd76482c67a2e21f2796db166a2/src/main/java/com/puppycrawl/tools/checkstyle/checks/design/VisibilityModifierCheck.java#L245
`int` is not an immutable type, even if the enumeration was correctly identified as a final class. `Integer` would be the appropriate immutable type.

**Comment by scompt** [comment]
1) The field isn't static because there are actually different values for each enum case. This is the most simple case that reproduces my bug. The actual code looks something like the enum [here](http://tutorials.jenkov.com/java/enums.html#enum-fields).

2) int is a primitive, which is checked [elsewhere](https://github.com/checkstyle/checkstyle/blob/ba6781e1098fbbd76482c67a2e21f2796db166a2/src/main/java/com/puppycrawl/tools/checkstyle/checks/design/VisibilityModifierCheck.java#L636) and documented to be handled by this check:

> allowPublicImmutableFields - which allows immutable fields be declared as public if defined in final class. Default value is true
> Field is known to be immutable if:
> It's declared as final
> Has either a primitive type or instance of class user defined to be immutable (such as String, ImmutableCollection from Guava and etc)

**Comment by rnveach** [comment]
You are correct, I missed primitives.

**Comment by romani** [comment]
yes, this is a bug.

**Comment by romani** [comment]
fix is merged.
